File: infracam.py
# infracam.py
# Description
# Test programs for the infracam project
#
# History:
#   2014-09-01  gel first tests
#
# defined constants used in python wrappers of wiringPi
# INT_EDGE_SETUP    0
# INT_EDGE_FALLING  1
# INT_EDGE_RISING   2
# INT_EDGE_BOTH     3
# PUD_OFF           0
# PUD_DOWN          1
# PUD_UP            2

import wiringpi2
wiringpi2.wiringPiSetup()	# For sequential pin numbering
from array import *
import time

sequence = array('i', [262, 294, 330, 349, 392, 440, 494, 525, 0])
vbeep = [0.05, 0.05, 0.05, 0.2]
onebeep = [0.05]
twobeeps = [0.05, 0.05]

wiringpi2.pinMode(0,0)          # Pin 0 for switch, INPUT
wiringpi2.pullUpDnControl(0, 1) # set pull UP resistor for pin 0 switch
wiringpi2.pinMode(1,1)          # Pin 1 for beeper, OUTPUT
wiringpi2.softToneCreate(3)     # initialize softTone on Pin 3

def beeper(beeps):
	"Beep all of the requested beeps in the beeps list"
	for i in beeps:
		wiringpi2.digitalWrite(1,1)	# beep on
		time.sleep(i)								# length of beep from beeps list
		wiringpi2.digitalWrite(1,0)	# beep off
		time.sleep(0.025)						# interval between beeps

def gotButton():
  "Double beeps after every button press from interrupt call"
  # beeper(twobeeps)
  print("Got a button press")
  return True

# Main Program
wiringpi2.wiringPiISR(0, 2, gotButton)
k=3
while k>0 :
	for i in sequence:
		print(i)
		wiringpi2.softToneWrite(3, i)
		time.sleep(0.5)
	beeper(onebeep)
	k-=1
time.sleep(0.5)
beeper(vbeep)

File: infracam.c
/*
 * infracam.c
 * Description
 * Test programs for the infracam project
 *
 * History:
 *   2014-09-01  gel first tests
 *

 */
#include <stdio.h>
#include <string.h>
#include <errno.h>
#include <stdlib.h>
#include <wiringPi.h>
#include <softTone.h>

#define SWITCH    0       // GPIO pin for switch
#define BEEP      1       // GPIO pin for beeper
#define PIEZO     3       // GPIO pin for piezo speaker

/*
 * ******************************************************************************
 * local functions
 * ******************************************************************************
 */

/* beeper */
void beeper(int beeps [], int numbeeps) {
  int i;
  for (i = 0; i < numbeeps; i++ ) {
		digitalWrite(BEEP,1);	        // beep on
		delay(beeps[i]);              // length of beep from beeps list
		digitalWrite(BEEP,0);	        // beep off
		delay(25);                    // interval between beeps
  }
}

 /* myInterrupt: gotButton */

void gotButton(void) {
  int twobeeps [] = { 50, 50 } ;
  int numtwobeeps = sizeof(twobeeps) /sizeof(twobeeps[0]);
  printf ("Got a button press");
  beeper (twobeeps, numtwobeeps);
  return ;
}
/*
 *********************************************************************************
 * main
 *********************************************************************************
 */

int main (void) {
  int sequence [] = { 262, 294, 330, 349, 392, 440, 494, 525, 0 } ;
  int numsequence = sizeof(sequence) / sizeof (sequence[0]);
  int vbeep [] = { 50, 50, 50, 200 } ;
  int numvbeep = sizeof(vbeep) / sizeof(vbeep[0]);
  int onebeep [] = { 50 } ;
  int numonebeep = sizeof(onebeep) / sizeof(onebeep[0]);

  wiringPiSetup();                //For the original sequential pin numbering
  pinMode (SWITCH, INPUT);            // Pin for switch, INPUT
  pullUpDnControl (SWITCH, PUD_DOWN);   // set pull UP resistor for switch
  pinMode (BEEP, OUTPUT);             // beeper pin, OUTPUT
  softToneCreate(PIEZO) ;             // initialize softTone 

  wiringPiISR(SWITCH, INT_EDGE_RISING, &gotButton) ;
  int k=3 ;
  int i;

  while (k>0) {
	  for ( i=0 ; i<numsequence; i++ ) {
		  printf("%d\n", sequence[i]);
		  softToneWrite (PIEZO, sequence[i]);
		  delay(500);
    }
	  beeper (onebeep, numonebeep);
	  k--;
  }
  delay(500);
  beeper (vbeep, numvbeep);
  return 0;
}


